app/models/channels.js

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+export class Channels {
+    static schema = {
+        name: 'Channels',
+        primaryKey: 'id',
+        properties: {
+            id: 'string',
+            createAt: 'int',
+            updateAt: 'int',
+            deleteAt: {type: 'int', default: 0},
+            team: 'Teams',
+            type: {type: 'string', indexed: true},
+            displayName: 'string?',
+            name: 'string',
+            header: 'string?',
+            lastPostAt: {type: 'int', default: 0},
+            totalMsgCount: {type: 'int', default: 0},
+            purpose: 'string?',
+            groupConstrained: {type: 'bool', default: false},
+            members: 'ChannelMembers[]',
+            memberCount: {type: 'int', default: 0},
+        },
+    }
+}
+
+export class ChannelMembers {
+    get channel() {
+        if (this.channel?.length) {
+            return this.channel[0];
+        }
+
+        return null;
+    }
+
+    get notifyPropsAsJSON() {
+        try {
+            return JSON.parse(this.notifyProps);
+        } catch {
+            return null;
+        }
+    }
+
+    set notifyPropsFromJSON(props) {
+        this.notifyProps = JSON.stringify(props);
+    }
+
+    static schema = {
+        name: 'ChannelMembers',
+        primaryKey: 'id',
+        properties: {
+            id: 'string', // ${channelId}-{$userId},
+            channels: {type: 'linkingObjects', objectType: 'Channels', property: 'members'},
```

Comment:
Shouldn't this just be
```suggestion
            channels: 'Channel',
```
since a ChannelMember object only has one channel?

Reply:
in this case is `Channels` as `Channels` is the name of the schema, but as I said before going to change the schema names to singular

Reply:
The type of this is confusing me though. It looks like one ChannelMember can have multiple Channel objects which isn't the case

Reply:
Maybe look at the `Inverse Relationships` section in the docs https://realm.io/docs/javascript/latest